scheduler/src/cook/mesos/api.clj

```diff
@@ -768,6 +851,8 @@
     (doseq [{:keys [executable? extract?] :as uri} (:uris munged)
             :when (and (not (nil? executable?)) (not (nil? extract?)))]
       (throw (ex-info "Uri cannot set executable and extract" uri)))
+    (doseq [dataset (:datasets munged)]
```

Comment:
This code seems out of place. Shouldn't you be validating it before munging it?

Reply:
I want to do the munging as early as possible, so that all of the code is consistent (i.e. dataset and partition keys are strings instead of keywords.)